dali/test/dali_operator_test.h

```diff
@@ -175,34 +220,30 @@ class DaliOperatorTest : public ::testing::Test, public ::testing::WithParamInte
    * push data through it and call output verification routines.
    * Every RunTest call will set up its own pipeline.
    *
-   * @tparam OutputBackend
    * @param inputs all inputs to the pipeline
    * @param outputs placeholder for outputs from pipeline
    * @param operator_arguments arguments, with which the pipeline will be called
    * @param verify function, that will be used for test verification
    */
-  template<typename OutputBackend>
-  void
-  RunTest(const std::vector<TensorListWrapper> &inputs, std::vector<TensorListWrapper> &outputs,
-          const Arguments &operator_arguments, const Verify &verify) {
+  void RunTest(const std::vector<TensorListWrapper> &inputs,
+               std::vector<TensorListWrapper> &outputs, const Arguments &operator_arguments,
+               const Verify &verify) {
     // TODO(mszolucha) implement
   }
 
-
   /**
    * Convenient overload, for specific, single-input/single-output graph
    */
-  template<typename OutputBackend>
   void RunTest(const TensorListWrapper &input, TensorListWrapper &output,
                const Arguments &operator_arguments, const VerifySingleIo &verify) {
-    std::string output_backend = detail::BackendStringName<OutputBackend>();
+    std::string op_backend = detail::GetOpDevice(operator_arguments);
     const auto batch_size = input.has_cpu() ? input.cpu().ntensor() : input.gpu().ntensor();
     auto pipeline = CreatePipeline(batch_size, num_threads_);
     if (input) {
       AddInputToPipeline(*pipeline, input);
     }
-    auto outputs = RunTestImpl<OutputBackend>(*pipeline, operator_arguments, input ? true : false,
-                                              input.backend(), output_backend);
+    auto outputs = RunTestImpl(*pipeline, operator_arguments, input ? true : false,
+                                              op_backend, op_backend);
```

Comment:
To be rigorous you could allow for mixed operators as well by returning a pair (input_backend, output_backend) in GetOpDevice and using it here

Reply:
Yeah, but there is also a case of an operator that has inputs both on CPU and GPU, and both of those cases could be handled after merging #369.